### PR TITLE
[fork-network] Optimize amend-access-keys using in-memory tries.

### PR DESCRIPTION
**Pull request by robin-near**
https://near.zulipchat.com/#narrow/stream/297873-pagoda.2Fnode/topic/New.20betanet.20infra

### CODE REVIEW

File: core/store/src/config.rs [hunk 1]
```diff
@@ -121,6 +121,8 @@ pub struct StoreConfig {
 
     // TODO (#9989): To be phased out in favor of state_snapshot_config
     pub state_snapshot_compaction_enabled: bool,
+
+    pub disable_rocksdb_compression: bool,
```
Comment (nikurt) [inline core/store/src/config.rs:125]: Please add a comment.

File: core/store/src/db/rocksdb.rs [hunk 2]
```diff
@@ -528,7 +528,12 @@ fn rocksdb_column_options(col: DBCol, store_config: &StoreConfig, temp: Temperat
     opts
 }
 
-fn set_compression_options(opts: &mut Options) {
+fn set_compression_options(opts: &mut Options, disable_compression: bool) {
```
Comment (nikurt) [inline core/store/src/db/rocksdb.rs:531]: (unrelated to this line)
Please mention somewhere in fork-network comments, that you recommend disabling rocksdb compression and what kind of speed up can be expected.

**Review by nikurt — APPROVED** [review]

File: core/store/src/db/rocksdb.rs (hunk 2 above)
Comment (robin-near) [reply core/store/src/db/rocksdb.rs]: Actually, I tried without it and there's no real difference. Compression spins up more threads to do compression, using up more CPU, but there's enough CPU with 8 cores anyway.

File: core/store/src/config.rs (hunk 1 above)
Comment (robin-near) [reply core/store/src/config.rs]: Removed the flag.

**Pull request merged**